.github/workflows/publish-dist.yml

```diff
@@ -29,9 +28,11 @@ jobs:
       - name: Commit to dist repo
         run: |
           set -ex
+          mkdir dist-repo
           cd dist-repo
+          tar --strip-components=1 -xf ../cockpit-*.tar.xz
           git init
-          git add .
+          git add dist/ node_modules/ package-lock.json tools/debian/copyright
```

Comment:
An additional hardening step I did in my `node_modules` workflow: please run `find -name '.git*' -delete` through here to make sure we aren't getting anything weird.  You're protected from the really bad stuff in the root directory, since you explicitly specify the filenames to unpack, but you could still get some unexpected things like ignore and attribute files in the subdirs.

Reply:
Good point! It needs to be a bit stronger though, as `-delete` does not delete directories ("find: cannot delete '...’: Directory not empty"). So something like

     find -name '.git*' -print0 | xargs -0 --no-run-if-empty rm -r --verbose

But then I realized that we can just let tar not unpack them in the first place, with `--exclude`. I did that now.

Reply:
In my case, `find` failing to delete directories returned with an error status and aborted the workflow, which was also fine.  Your approach is also very nice, though. :) I might steal it :rofl:

Reply:
That's a good point - in fact, let's not even delete, but just fail the workflow if there is any ".git*" stuff inside. Because that's either a trap, or a bug of our's.

Reply:
Updated to hard-fail on any top-level `.git*` stuff. We *may* need to dial that back at some point if needed, but until then, better safe than sorry?